web/app/components/AccountMenu.tsx

```diff
@@ -0,0 +1,100 @@
+import { LinkOff } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+  Stack,
+  Typography,
+} from "@mui/material";
+import React, { FC, useState } from "react";
+import { useEthereum } from "~/context/EthereumContext";
+import { useVoter } from "~/context/VoterContext";
+import { formatAccountString } from "~/lib/utils";
+
+export const AccountMenu: FC = () => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const { account, logoutEthereumAccount, loading } = useEthereum();
+  const { voter } = useVoter();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <>
+      <Button
+        id="account-menu"
+        aria-controls={open ? "account-menu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        onClick={handleClick}
+        color="neutral"
+      >
+        <Stack direction="row" spacing={1}>
+          <Typography>
+            {voter && account ? `${voter.firstName} ${voter.lastName}` : ""}
```

Comment:
Avec React quand tu veux rien render, return `null`

```suggestion
            {voter && account ? `${voter.firstName} ${voter.lastName}` :  null}
```